src/diffusers/pipelines/flux/pipeline_flux_controlnet.py

```diff
@@ -429,6 +451,30 @@ def check_inputs(
         elif prompt_2 is not None and (not isinstance(prompt_2, str) and not isinstance(prompt_2, list)):
             raise ValueError(f"`prompt_2` has to be of type `str` or `list` but is {type(prompt_2)}")
 
+        if (
```

Comment:
I think we can remove almost all of the checks and only do following two things:
1. make sure if it is a multi controlnet, image is a list with same length of number of controlnet
2. add a check here https://github.com/huggingface/diffusers/blob/56d6d21baeb203a249b3049835916bfb0bfad061/src/diffusers/pipelines/controlnet/pipeline_controlnet_sd_xl.py#L862
```python
        if image_batch_size == 1:
            repeat_by = batch_size
        elif image_batch_size == batch_size:
            # image batch size is the same as prompt batch size
            repeat_by = num_images_per_prompt
        else:
             raise ValueError("...")
   ```

it should be sufficient no? would we miss anything here?